### PR TITLE
Added ms oauth 2 grant flow

### PR DESCRIPTION
**Pull request by d-co-white**
Ms oauth 2 auth grant flow

### CODE REVIEW

File: configs/authentication/stackhawk-ms-oauth2-auth-code.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    script:
+      name: ms-oauth2-auth-code.kts
+      # Full explanation of parameters and credentials here
+      # https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow
+      credentials:
+        username: ${SCAN_USER}
+        password: ${SCAN_PASS}
+        clientId: ${CLIENT_ID}
+        client_secret: ${CLIENT_SECRET}
+        redirect_url: "http%3A%2F%2Flocalhost%2Fmyapp%2F" # Change to your redirect uri (url encoded)
```
Comment (kcberg) [inline configs/authentication/stackhawk-ms-oauth2-auth-code.yml]: Do these need to be encoded?

Comment (d-co-white) [reply configs/authentication/stackhawk-ms-oauth2-auth-code.yml]: It's in the request url, so yes?

**Review by kcberg — CHANGES_REQUESTED** [review]
encoding questions

File: configs/authentication/stackhawk-ms-oauth2-auth-code.yml (hunk 1 above)
Comment (kcberg) [reply configs/authentication/stackhawk-ms-oauth2-auth-code.yml]: can't we encoded it in the script?

Comment (d-co-white) [reply configs/authentication/stackhawk-ms-oauth2-auth-code.yml]: Yeah we can.  I'll go do that!

**Review by kcberg — APPROVED** [review]

**Pull request merged**